client/scripts/mifare_ad.lua

```diff
@@ -0,0 +1,186 @@
+
+-- Ability to read what card is there
+local getopt = require('getopt')
+local cmds = require('commands')
+local taglib = require('taglib')
+
+local desc = 
+[[This script will automatically check Mifare cards for MADs
+(Mifare Application Directory)
+
+Arguments:
+	-d 				debug logging on
+	-h 				this help
+
+]]
+local example = "script run xxx"
+local author = "Mazodude"
+---
+-- PrintAndLog
+function prlog(...)
+	print(...)
+end
+--- 
+-- This is only meant to be used when errors occur
+function oops(err)
+	prlog("ERROR: ",err)
+	return nil,err
+end
+
+--- 
+-- Usage help
+function help()
+	prlog(desc)
+	prlog("Example usage")
+	prlog(example)
+end
+
+function debug(...)
+	if DEBUG then 
+		prlog("debug:", ...)
+	end
+end
+
+
+local function show(data)
+	if DEBUG then
+	    local formatString = ("H%d"):format(string.len(data))
+	    local _,hexdata = bin.unpack(formatString, data)
+	    debug("Hexdata" , hexdata)
+	end
+end
+--- Fire up a connection with a tag, return uid
+-- @return UID if successfull
+-- @return nil, errormessage if unsuccessfull
+local function open()
```

Comment:
How about using the lualibs functions;

	-- GET TAG UID
	tag, err = lib14a.read(false, true)
	if not tag then return oops(err) end
core.clearCommandBuffer()

Reply:
you will need 
local lib14a = require('read14a')